### PR TITLE
new custom table for my capabilities

### PR DESCRIPTION
**Pull request by HrisyToteva**
https://github.com/dfds/cloudplatform/issues/1970

Replaced the old React Table with a custom one that has all special functionalities disabled (except sorting) in My Capabilities.

image

### CODE REVIEW

File: src/src/pages/capabilities/MyCapabilities.js [hunk 1]
```diff
@@ -68,6 +70,89 @@ export default function MyCapabilities() {
...
+
+      return copy;
+
+    });
+
+
+
+    setFullTableData(tableData);
+
+  }, [isLoading, appStatus, isLoadingCosts])
```
Comment (kralle333) [inline src/src/pages/capabilities/MyCapabilities.js:153]: guess we also need `isLoadingAwsResourcesCounts` here or solve this issue in a nicer way (figure out how to managing loading of capability metrics non-boolean way)

**Review by kralle333 — COMMENTED** [review]
Only have a small comment, but otherwise it looks good to me! I would like @andreasfrisch to give it a more thorough look as he more knowledgable of frontend.

File: src/src/pages/capabilities/MyCapabilities.js (hunk 1 above)
Comment (andreasfrisch) [reply src/src/pages/capabilities/MyCapabilities.js:153]: If we want to show spinners for individual parts, then I am afraid we will need boolean flags for each such part.
For now our amount of flags is fairly manageable, but if it grows too big let us look at this.

Crazy thought of the day; Having a single key-value map of listeners, triggering on changes, and dispatching calls to the relevant entities -- though I am not sure this is easier to maintain 😅

File: src/src/pages/capabilities/myCapabilities.module.css [hunk 2]
```diff
@@ -5,7 +5,6 @@
 }
 
 .costs{
-  width: 100%;
```
Comment (andreasfrisch) [inline src/src/pages/capabilities/myCapabilities.module.css:8]: We are slowly getting away from the world where table content is responsible for knowing how the table works. This is nice!

File: src/src/pages/capabilities/MyCapabilities.js [hunk 3]
```diff
@@ -68,6 +70,89 @@ export default function MyCapabilities() {
...
+        header: 'Aws Resources',
+        size: 150,
+        enableColumnFilterModes: false,
+        muiTableHeadCellProps: {
+          align: 'center'
+        },
+        muiTableBodyCellProps: {
+          align: 'center'
+        }
+
```
Comment (andreasfrisch) [inline src/src/pages/capabilities/MyCapabilities.js:103]: I will suggest using the `InlineAwsCountSummary` in a Cell-definition as for the other columns.
Firstly this is already defined and does some simple rendering with a logo, secondly we offload this rendering to a specialized component just as we do for costs.

This is not necessary at this point, though

**Review by andreasfrisch — APPROVED** [review]
This is nice 🎉 
I have some suggestions, but none important enough to block this giant step towards the future (tm)

File: src/src/pages/capabilities/MyCapabilities.js [hunk 4]
```diff
@@ -68,6 +70,89 @@ export default function MyCapabilities() {
   useEffect(() => {
     setShowAwsResourcesSpinner(isLoadingAwsResourcesCounts);
   }, [isLoadingAwsResourcesCounts]);
+
+
+  const columns = useMemo(
```
Comment (andreasfrisch) [inline src/src/pages/capabilities/MyCapabilities.js:75]: This is not something we can do anything about and the `useMemo` is part of the official documentation too.
It just feels funny, that we are passing something a function and then insisting on the result never changing.
Why have the function in the first place 🤔 
Oy vey; I am probably just missing some fascinating big-picture things

File: src/src/pages/capabilities/MyCapabilities.js [hunk 5]
```diff
@@ -90,65 +186,53 @@ export default function MyCapabilities() {
...
+              }}
+              muiTableBodyCellProps={{
+                sx: {
+                  fontWeight: '400',
+                  fontSize: '16px',
+                  fontFamily: 'DFDS',
+                  color: '#4d4e4c',
+                  padding: '5px',
+                  // margin: 0,
+                  // minHeight: 0,
```
Comment (andreasfrisch) [inline src/src/pages/capabilities/MyCapabilities.js:206]:
```suggestion
```

File: src/src/pages/capabilities/MyCapabilities.js [hunk 6]
```diff
@@ -90,65 +186,53 @@ export default function MyCapabilities() {
...
+                },
+              }}
+              muiTableBodyCellProps={{
+                sx: {
+                  fontWeight: '400',
+                  fontSize: '16px',
+                  fontFamily: 'DFDS',
+                  color: '#4d4e4c',
+                  padding: '5px',
+                  // margin: 0,
```
Comment (andreasfrisch) [inline src/src/pages/capabilities/MyCapabilities.js:205]:
```suggestion
```

**Review by andreasfrisch — APPROVED** [review]
Too minor suggestions; Take them or leave them :)

**Pull request merged**